augur/filter.py

```diff
@@ -788,10 +788,18 @@ def apply_filters(metadata, exclude_by, include_by):
     for filter_function, filter_kwargs in exclude_by:
         # Apply the current function with its given arguments. Each function
         # returns a set of strains that passed the corresponding filter.
-        passed = metadata.pipe(
-            filter_function,
-            **filter_kwargs,
-        )
+        try:
+            passed = metadata.pipe(
+                filter_function,
+                **filter_kwargs,
+            )
+        except Exception as e:
+            if filter_function.__name__ == 'filter_by_query':
+                if type(e) == pd.core.computation.ops.UndefinedVariableError:
+                    raise AugurError(f"Query contains a column that does not exist in metadata.")
+                raise AugurError(f"Error when applying query. Ensure the syntax is valid per https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#indexing-query.")
```

Comment:
Adding a "from" clause to each raise will make the exception chaining explicit and intentional instead of being reported as an unexpected error from within the exception handler.

```suggestion
                    raise AugurError(f"Query contains a column that does not exist in metadata.") from e
                raise AugurError(f"Error when applying query. Ensure the syntax is valid per <https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#indexing-query>.") from e
```

I also surrounded the URL in angle brackets, which is plain text convention where surrounding chars may not be properly included/excluded (e.g. the trailing period).